### DISCUSSION TITLE
Should we release 0.67.4?

### DISCUSSION THREAD

**Discussion by ShikaSD** [reactwg/react-native-releases#15] [Patches]
RELEASED - please refer to https://github.com/reactwg/react-native-releases/discussions/18 for 0.67.5

----

 Current Release: [0.67.3](https://github.com/facebook/react-native/releases/tag/v0.67.3)
    
Conversations on this thread are limited: 
* [major release issues](https://github.com/facebook/react-native/wiki/Release-FAQ#what-is-release-blocking)
* [qualified cherry-pick requests](https://github.com/facebook/react-native/wiki/Release-FAQ#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://github.com/facebook/react-native/wiki/Release-FAQ#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).
    
Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)
    
In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue. 
If the issue is a [major release issue](https://github.com/facebook/react-native/wiki/Release-FAQ#what-is-release-blocking), please reference the issue here.
    
---
#### List of qualified picks
 
1. https://github.com/facebook/react-native/commit/8d50bf11335
    
#### Local commits to backport to main
1.

**Comment by mikehardy** [comment]
Copied from https://github.com/reactwg/react-native-releases/discussions/13#discussioncomment-2267363
----
Just saw this in react-native weekly digest - seemed like this affects back to the 0.66 tree and was finally merged, but recently enough it is not on the pick list here? https://github.com/facebook/react-native/commit/8d50bf11335 - fixes a regression in Switch behavior. I'll post similar comments on 0.67 / 0.66 release planning threads but @d4vidi / @asafkorem / @jonathanmos might do well to keep eyes on it :eyes: - my comments are intended to bridge people into this maybe-pick-it conversation but I am likely not most aware person for this change

**Comment by aprilmintacpineda** [comment]
Hi there, just wanna reach out to ask, because seems no one else is experiencing this or something, https://github.com/facebook/react-native/issues/33165 does it ring a bell to anyone?

**Reply by cortinico** [reply]
This is offtopic:

> In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issue](https://github.com/facebook/react-native/wiki/Release-FAQ#what-is-release-blocking), please reference the issue here.

If the issue gets fixed, you can ask for it to be backported to a potential 0.67.4. But that's not the case at this stage.

**Comment by xzilja** [comment]
Update causes following build errors on ios (Updated from 0.67.3). Expanding on these errors doesn't give any extra info I'm afraid. I also tried

- Removing Podfile.lock, Pods and .xcworkspace
- Removing DerrivedData folder
- Re-installing pods and re-generating xcworkspace
- Cleaning build

Error still persists. Downgrading back to 0.67.3 solves it.

This happens in latest XCode version / M1 Mac

<img width="543" alt="Screenshot 2022-03-18 at 11 55 46" src="https://user-images.githubusercontent.com/3154053/158994254-3c1c21ee-2b84-4794-a522-4f4365cf58c1.png">

**Reply by leotm** [reply]
deffo had ambiguous PhaseScriptExecution headache in the past too so i feel ya
seems similar [here](https://github.com/facebook/react-native/issues/31181#issuecomment-1063157109) on slightly behind Xcode/Monterey versions

just confirming ^ Xcode 13.3 right (March 14, 22)
i did the 10gb upgrade too from 13.3 RC (March 8, 22) this wk
and Monterey 12.3 maybe (my version atm)

might be able to test this wkend on my M1 MBP (2020)

edit: btw is your `which node` or `whereis node` living at (or symlinked) `/opt/homebrew/bin/node` too?
some people had trouble with _nvm_ before, i'm on node v17.7.1 (i think node location might be related)

**Reply by xzilja** [reply]
Yep 13.3
I also tried all common fixes suggested from basic googling around node versions / paths even though they didn't apply in my case (not using nvm or homebrew). Don't have any weird symbols not spaces in my paths as well. Also worth noting when testing - I had a weird cache bug where even though I downgraded back to rn 0.67.3 and deleted Podfile.lock I was still installing pods from 0.67.4 (which were causing this error), so make sure you are back on 0.67.3 on all your pod versions as well.

**Reply by tido64** [reply]
You should be able to expand the line by clicking on the button to the right:
![image](https://user-images.githubusercontent.com/4123478/159019306-710045f8-e7a5-402f-995c-6a8f060266c2.png)

**Reply by xzilja** [reply]
Yep, did that, all info thats there is a generic message about `PhaseScriptExecution` and a long list of exports for various env vars. I scrolled through it several times to see if anything specific pops out, but there was nothing. Will try to post it here fully if no one else manages to replicate.

**Reply by leotm** [reply]
i remember was a toughie extracting decent debug info
i'll upgrade my [template](https://github.com/leotm/react-native-template-typescript) later 0.68.0-rc.2 -> 0.68.0-rc.3
and try 0.68.0-rc.2 -> 0.67.3 -> 0.67.4 next

**Reply by mikehardy** [reply]
I'll check 0.67.4 soon as well, and I have an M1 with Xcode 13.3 / Monterey 12.3 updated now, @IljaDaderko I can say that if you copy that entire set of env vars *and* the script being run, and run it in Terminal, usually you get the correct error message, and you can start digging in to that script (even if it's in a temporary build folder) and adding troubleshooting output prints to it. That's how I've zeroed in on problems like this before.

**Reply by mikehardy** [reply]
0.67.4 works for me on android + ios debug and release +hermes +flipper on a real app release to the stores, from an *intel* mac, Xcode 13.3 / macOS 12.3. I understand this appears it could be an arm64 problem but I don't have access to the M1 just yet. x86_64 at least does not appear broken

**Reply by mikehardy** [reply]
Got my hands on the M1. Everything worked fine for me, Xcode 13.3 / macOS 12.3, ios debug +hermes +flipper - didn't check release but that wasn't specified. works for me?

**Reply by xzilja** [reply]
@mikehardy Alright, thank you for checking this out. Sounds like the issue is specific to me (eh..) Will have to investigate further. Wonder if following things can have an effect on this:

1. I recently updated to latest cocoapods version (1.11.3)
2. My nodejs / cocoapods are installed through homebrew

I'll try your suggested approach of running that command manually from terminal as well 👌 hopefully that gives more concrete error.

**Reply by xzilja** [reply]
Well after a lot of fighting, my issue was resolved by completely deleting the repo and re-pulling it again from github. I was then able to update. Still no clue what was the issue last time though.

**Reply by leotm** [reply]
great stuff ^ [`git -fdx`](https://github.com/reactwg/react-native-releases/discussions/13#discussioncomment-2266915) helped me [recently](https://github.com/software-mansion/react-native-screens/issues/1354#issuecomment-1059135519) too and [upgrading to new arch](https://github.com/reactwg/react-native-releases/discussions/13#discussioncomment-2269256) a ton

roughly equivalent to delete -> re-clone i think/hope

**Reply by leotm** [reply]
oh and mb [react-native-clean-project](https://github.com/pmadruga/react-native-clean-project) as well ^
@mikehardy noticed you helping there

and everyone else in [troubleshooting guide](https://github.com/facebook/react-native/issues/31941) / various other projects

happy to sponsor you guys (when paperwork lands for my next role 🤞)

p.s. what we've learnt here and [past](https://github.com/react-native-async-storage/async-storage/issues/656#issuecomment-898274977) is a good night's sleep always seems to be the solution
